cpp/arcticdb/async/async_store.hpp

```diff
@@ -404,65 +404,47 @@ class AsyncStore : public Store {
         const std::shared_ptr<DeDupMap> &de_dup_map,
         const BatchWriteArgs &args) override {
```

Comment:
I think we should get rid of `BatchWriteArgs`. This method is only called in one place, which doesn't provide the argument:
https://github.com/man-group/ArcticDB/blob/master/cpp/arcticdb/pipeline/write_frame.cpp#L108
It would make more sense to me to use `follw::window` for both the encoding and writing tasks, and to use `cpu_thread_count()` and `io_thread_count()` respectively as the window sizes. That should ensure maximum performance, while at the same time keeping the number of tasks in the queues short if there are multiple callers of this function at the same time.